src/middleware.py

```diff
@@ -0,0 +1,25 @@
+from django.conf import settings
+from django.contrib.auth import login
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import FacilityUser
+
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:
+    MiddlewareMixin = object
+
+
+class AlwaysAuthenticatedMiddleware(MiddlewareMixin):
+    def __init__(self, *args, **kwargs):
+
+        self.username = "endless"
+        super(AlwaysAuthenticatedMiddleware, self).__init__(*args, **kwargs)
+
+    def process_request(self, request):
+        if not request.user.is_authenticated():
+            facility = Facility.get_default_facility()
+            user, created = FacilityUser.objects.get_or_create(
+                username=self.username, facility=facility
+            )
+            user.backend = settings.AUTHENTICATION_BACKENDS[0]
+            login(request, user)
```

Comment:
It bugs me that this user isn't an administrator. If our wizard for installing content works with this change, it probably _shouldn't_ work :b

If we don't want this to be an admin user - which I can understand, as the facility settings don't make a whole lot of sense here - maybe it could set that extra `can_manage_content` permission like we do in kolibri-desktop-auth-plugin? https://github.com/endlessm/kolibri-desktop-auth-plugin/blob/a655f73c9940eff7f182008abafb421090aa8cec/kolibri_desktop_auth_plugin/models.py#L61-L65. That should avoid any sneaky permissions issues.

Reply:
The install content button in the explore plugin is working because it uses a custom view, so the operations are done in the backend and the authentication is not needed. But in any case, it's a good idea to add that permissions, so we can extend that option and maybe use default content interface in the future, instead of a custom API.